swatch-core/src/main/java/org/candlepin/subscriptions/registry/TagProfile.java

```diff
@@ -111,6 +114,8 @@ private void handleTagMapping(TagMapping mapping) {
       roleToTagLookup
           .computeIfAbsent(mapping.getValue(), k -> new HashSet<>())
           .addAll(mapping.getTags());
+    } else if ("arch".equals(mapping.getValueType())) {
+      archToTagLookup.put(mapping.getValue(), mapping.getTags().stream().findFirst().orElseThrow());
```

Comment:
It feels like the mapping here should be Map<String, Set<String>>. I realize that the code that uses this grabs the first tag it finds for the arch, but if we ever need the entire list for something we would need to change this. Perhaps it would be better to have the code that uses this mapping decide whether it wants the first item.

I'd also like to see a test around this code.